File: app/src/main/java/com.example.safehome/Wall/Coordinates.java
package com.example.safehome.Wall;

public class Coordinates {
    int x;
    int y;

    /**
     * Defines a coordinate system.
     * @param x x coordinate
     * @param y y coordinate
     */
    public Coordinates(int x, int y) {
        this.x = x;
        this.y = y;
    }

    /**
     * Get the x coordinate.
     * @return the x coordinate
     */
    public int getX() {
        return x;
    }

    /**
     * Set the x coordinate
     * @param x the x coordinate
     */
    public void setX(int x) {
        this.x = x;
    }

    /**
     * Get the y coordinate
     * @return the y coordinate
     */
    public int getY() {
        return y;
    }
    /**
     * Set the y coordinate
     * @param y the y coordinate
     */
    public void setY(int y) {
        this.y = y;
    }



}

File: app/src/main/kotlin/com.example.safehome/Wall/Coordinates.kt
package com.example.safehome.Wall

import org.slf4j.LoggerFactory
import java.lang.Exception

class Coordinates(x: Int?, y: Int?) {
    /**
     * Get the x coordinate.
     * @return the x coordinate
     */
    /**
     * Set the x coordinate
     * @param x the x coordinate
     */
    var x: Int? = null
    /**
     * Get the y coordinate
     * @return the y coordinate
     */
    /**
     * Set the y coordinate
     * @param y the y coordinate
     */
    var y: Int? = null

    /**
     * Defines a coordinate system.
     * @param x x coordinate
     * @param y y coordinate
     */
    init {
        try {
            check(!(x == null || y == null)) { "Parameter \"x and y\" are required." }
            this.x = x
            this.y = y
            log.info("{}", this)
        } catch (e: Exception) {
            log.error("{}", e)
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(Coordinates::class.java)
    }
}